rxbinding-support-v4/src/androidTest/java/com/jakewharton/rxbinding/support/v4/widget/RxSwipeRefreshLayoutTestActivity.java

```diff
@@ -18,15 +21,21 @@
   private final Runnable stopRefreshing = new Runnable() {
     @Override public void run() {
       swipeRefreshLayout.setRefreshing(false);
-      handler.postDelayed(this, 500);
     }
   };
 
   @Override protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
 
     swipeRefreshLayout = new SwipeRefreshLayout(this);
-    swipeRefreshLayout.setId(1);
+    swipeRefreshLayout.setId(R.id.swipe_refresh_layout);
+    swipeRefreshLayout.setOnTouchListener(new View.OnTouchListener() {
+      @Override public boolean onTouch(View v, MotionEvent event) {
+        handler.removeCallbacks(stopRefreshing);
```

Comment:
Please!

On Fri, Jun 24, 2016 at 5:47 PM Ivan Škorić <notifications@github.com>
wrote:

> In
> rxbinding-support-v4/src/androidTest/java/com/jakewharton/rxbinding/support/v4/widget/RxSwipeRefreshLayoutTestActivity.java
> <https://github.com/JakeWharton/RxBinding/pull/261#discussion_r68467910>:
>
> >      }
> >    };
> >
> >    @Override protected void onCreate(Bundle savedInstanceState) {
> >      super.onCreate(savedInstanceState);
> >
> >      swipeRefreshLayout = new SwipeRefreshLayout(this);
> > -    swipeRefreshLayout.setId(1);
> > +    swipeRefreshLayout.setId(R.id.swipe_refresh_layout);
> > +    swipeRefreshLayout.setOnTouchListener(new View.OnTouchListener() {
> > +      @Override public boolean onTouch(View v, MotionEvent event) {
> > +        handler.removeCallbacks(stopRefreshing);
>
> Yes, we could do it on ACTION_UP as well with shorter delay, that way we
> also get flexibility for swipe speeds. Want me to do it that way?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/JakeWharton/RxBinding/pull/261/files/6142f892f592924499bd3ff84ad576c157c86daa#r68467910>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAEEEcmBdV2I2du0hFcDd2e3qQmvDIL_ks5qPFBogaJpZM4I-JRH>
> .
>